project-api/ballerina-projects/src/main/java/io/ballerina/projects/writers/exceptions/NoPermissionException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.projects.writers.exceptions;
+
+/**
+ * {@code NoPermissionException} throws when permission denied for certain action.
+ *
+ * @since 2.0.0
+ */
+public class NoPermissionException extends RuntimeException {
```

Comment:
We can use IOExceptions instead of this custom exception.

Reply:
Removed this custom exception